### ISSUE TITLE
add horizontal / collapsible level switcher

### ISSUE THREAD

**Issue by smallTrogdor** [SchweizerischeBundesbahnen/journey-maps-client-flutter#61]
### Description

Implement the horizontal and collapsible FloorSwitcher.

### Motivation

The current floor switcher takes up a lot of space of the screen when not necessarily being needed. The new horizontal and collapsible floor switcher is much nicer.

### Proposed Solution (optional)

- Make a new floor switcher HorizontalFloorSwitcher

### Design

<img width="1882" height="1077" alt="Image" src="https://github.com/user-attachments/assets/2e039df9-539f-4de2-8cdf-aa6cc8db84b4" />

**Comment by smallTrogdor** [comment]
## Problem Statement

The floor switcher is a fixed 44 px vertical column of tiles pinned to the top right of
the map, and it is always expanded. At a station with five floors it occupies roughly a
third of the screen height, permanently, whether or not the traveller has any interest in
changing floor. The control that exists to help you read the map is covering the map.

The cost lands hardest exactly where the feature matters most: large multi-level stations
have the most floors, so the switcher is largest precisely where the traveller most needs
to see the plan underneath it. A consumer's only escape today is to abandon the default
UI, supply a custom builder, and position the control themselves — which does not make it
smaller, only differently placed.

## Solution

Add a second floor switcher variant that is horizontal and collapsible, and make it
selectable alongside the existing one.

Collapsed, it is a single circular control the size of the other map buttons. Tapping it
expands a stadium-shaped pill leftwards, revealing the available floors in a row with a
close affordance where the circle was. Picking a floor collapses it again. When nothing is
selected the collapsed circle shows a floors icon; once a floor is chosen it shows that
floor, so the traveller can see their current level at a glance without the control taking
any more room than a button.

The existing vertical control keeps working unchanged. Consumers opt into the horizontal
one through a new orientation setting, and it becomes the default in 3.0.0. Because the
package's widget names have always said "selector" while its behaviour has always said
"switch", this is also the moment the vocabulary is unified on **floor switcher**, with
the old names preserved as deprecated aliases so nothing breaks in 2.x.

## User Stories

1. As a traveller using a map in a large station, I want the floor switcher to stay out of
   the way until I need it, so that I can see the station plan I opened the map for.
2. As a traveller, I want to expand the floor switcher with a single tap, so that changing
   floor is not slower for having been collapsed.
3. As a traveller, I want the control to collapse again as soon as I pick a floor, so that
   I do not have to perform a second dismissing action to get my map back.
4. As a traveller, I want to close the expanded switcher without changing floor, so that I
   can back out of an interaction I opened by accident.
5. As a traveller, I want to see which floor I am currently on without expanding anything,
   so that I can orient myself at a glance.
6. As a traveller who has not chosen a floor yet, I want the control to show a recognisable
   floors icon, so that I can tell what it does before I have ever used it.
7. As a traveller, I want the currently selected floor to stand out when the switcher is
   open, so that I can see where I am relative to the floor I am about to pick.
8. As a traveller, I want to deselect my current floor, so that I can return to the default
   view of the map.
9. As a traveller, I want the switcher to expand out from the point I touched, so that the
   close affordance lands under the finger that opened it.
10. As a traveller, I want floors ordered in a way that matches how I read, so that I do
    not have to hunt for the floor I want.
11. As a traveller, I want expansion to be animated rather than instantaneous, so that I
    can see where the control came from and what just happened.
12. As a traveller who has panned to a different station, I want the switcher to reset
    rather than showing floors that are no longer there, so that I am not offered a choice
    that does nothing.
13. As a traveller, I want the switcher to disappear entirely where the map has no indoor
    data, so that the screen is not cluttered with a control that has nothing to offer.
14. As an app developer, I want to choose between a vertical and a horizontal floor
    switcher, so that I can pick the one that suits my app's layout.
15. As an app developer, I want the horizontal switcher available through the default UI
    and not only through a custom builder, so that I can adopt it without rebuilding my
    map's chrome.
16. As an app developer, I want the vertical switcher to remain the default in 2.x, so that
    upgrading within the current major version changes nothing I did not ask for.
17. As an app developer, I want a release cycle to evaluate the horizontal switcher before
    it becomes the default, so that the 3.0.0 upgrade holds no surprises.
18. As an app developer, I want both switchers available in a compact size, so that the
    control matches the small map controls I already use.
19. As an app developer, I want to supply my own floor labels, so that I can show "EG" and
    "UG" to Swiss travellers instead of raw integers.
20. As an app developer, I want floor labels supplied the same way on both switcher
    variants, so that switching orientation does not change how I localize.
21. As a developer building for a non-German-speaking region, I want the package to hold no
    opinion about floor naming, so that I can translate through my own pipeline.
22. As an app developer with negative floors, I want to label basements meaningfully, so
    that travellers see "UG" rather than "-1".
23. As an app developer who has themed the floor switcher, I want my theming to apply to
    the horizontal variant too, so that adopting it costs me no styling work.
24. As an app developer, I want to colour the switcher's icons independently of its floor
    labels, so that the affordance glyphs can be quieter than the numbers.
25. As an app developer, I want the rename to reach me as deprecation warnings rather than
    compile errors, so that I can migrate on my own schedule.
26. As an app developer, I want the deprecation messages to name the replacement and the
    version it disappears in, so that I can plan the migration without reading a changelog.
27. As an app developer upgrading to 3.0.0, I want the migration to be a mechanical rename,
    so that a major version bump does not become a project.
28. As a maintainer, I want the widget names to agree with the controller's vocabulary, so
    that nobody has to remember that the thing called a selector is operated by switching.
29. As a maintainer, I want the horizontal switcher's behaviour covered by tests, so that
    its expansion and selection logic cannot regress silently.
30. As a maintainer, I want a reusable harness for mounting map UI controls in tests, so
    that the next widget test does not have to reinvent the scaffolding.
31. As a maintainer, I want the vertical switcher untouched beyond its rename, so that a
    release billed as backward compatible actually is.
32. As a reviewer, I want the mechanical rename separable from the new widget, so that I
    can review the behavioural change without it being buried in renames.

## Implementation Decisions

Recorded in full in the architectural decision record covering this work; summarised here.
Where this spec and that record disagree, the record wins.

### Vocabulary and renaming

- **"Switcher" becomes the single term.** The existing floor selector, its compact
  variant, and its style extension are all renamed to floor switcher equivalents, with the
  existing control becoming the **vertical floor switcher**. This resolves a long-standing
  split: the controller's verb, the map's feature flag, the README's feature table and a
  dangling documentation reference all already said "switch".
- **Every renamed public name survives as a deprecated alias**, worded to match the
  package's existing deprecation style: name the replacement, state the version the
  deprecation lands after, and state that removal happens in 3.0.0. Types that are not
  exported from the package barrel are renamed without aliases.
- **Deprecation is not itself a breaking change.** It ships as a feature, cutting a minor
  release. Removal of the aliases happens in 3.0.0.

### Floor labels

- Both switcher variants accept a consumer-supplied **floor label builder**, mapping a
  floor to the string shown for it, defaulting to the integer's string form so existing
  output is unchanged. The package ships no localization and takes no position on floor
  naming; the "EG" in the design is the consumer's label for floor zero. This also covers
  negative floors.

### The horizontal floor switcher

- **Collapsible by definition.** There is no flag to make it non-collapsible and no flag to
  make it start open. It always starts collapsed.
- **States:**

  | Current floor | Collapsed shows | Expanded right-hand element |
  | --- | --- | --- |
  | none | a floors icon (layers with arrows) | a close icon (cross) |
  | selected | the floor's label, filled | a close icon (cross) |

- **Transitions:** tapping the collapsed circle expands; tapping the close icon collapses;
  selecting a floor switches to it and collapses; selecting the floor that is already
  current deselects it, resets the map to its default level, and collapses back to the
  icon state.
- **The deselect toggle is preserved from the vertical variant**, and is only expressible
  here because "no floor selected" now has a distinct visual form. Previously the
  distinction between no selection and ground level had no representation.
- **Layout:** floors run ascending left to right, which means reversing the controller's
  descending floor list for display. The pill is anchored to its right edge and grows
  leftward. The close icon occupies exactly the position the collapsed circle held, so the
  control expands out from the point of touch.
- **The icon end caps are not floor tiles.** The collapsed icon and the close icon are
  full-height circular caps matching the existing map icon buttons in both sizes, rather
  than inheriting the padded inner geometry of a floor tile. This keeps their tap targets
  at full control height.
- **Selected floor rendering:** bold and filled, reusing the selected background and text
  colours the vertical variant already defines.
- **Animation:** the expand and collapse animate over the same duration the existing floor
  tiles already use, with an ease-in-out curve. Width animates and the floor row is clipped
  by the animating pill; there is no separate fade, because a fade plus a width tween reads
  as two competing animations.
- **Floor-set changes collapse the control.** The available floors are recomputed on every
  camera idle, so the comparison must be **by value, not by identity** — an equal but
  rebuilt list must not disturb a traveller mid-interaction. The floor controller emits a
  single undifferentiated change notification for both floor-set and selection changes, so
  the widget caches the previous list and compares it itself.
- **Expansion state resets when the floor list empties.** The existing guard that renders
  nothing when no floors are available is retained, but internal expansion state must reset
  too, or the control reappears already open.
- **A compact variant ships alongside**, following the package's established convention of
  a separate class per size rather than a size parameter.

### Styling

- **One theme extension continues to serve both variants.** It gains exactly one new token,
  an icon colour for the affordance glyphs, defaulting to the same value as the text colour
  so existing themes are unaffected. The pill background, the dark-mode ring and the filled
  selection all reuse tokens the vertical variant already defines, so a consumer who has
  themed the floor switcher gets the horizontal one themed for free.

### Integration

- **A new orientation setting on the map** selects the variant, defaulting to vertical, and
  crosses with the existing compact-controls flag to give four combinations in the default
  UI. It flips to horizontal in 3.0.0.
- **The default UI keeps its existing spacing above the switcher**, and the horizontal
  branch is additionally aligned to the right edge so that the pill grows leftward instead
  of centring itself in the column of controls.

### Sequencing

- **Blocked on the typeface removal.** The package is dropping the SBB Web typeface, after
  which map text inherits the host application's typography. The horizontal switcher's
  labels should be written against the final typography rules rather than retrofitted.

## Testing Decisions

- **A good test here asserts what a traveller can observe** — what the control shows, what
  it does when tapped, what it asks the controller to do — not how the widget tree is
  assembled. Assertions about internal builders, private state fields, or the presence of
  particular layout widgets are implementation details and are avoided.
- **One seam: a shared harness for mounting map UI controls.** A floor switcher reads its
  controller from the map's UI container and its colours from theme extensions injected by
  the map's style container, so any widget test must mount both. A single reusable helper
  provides that wrapper with injectable stand-ins, and is the only new test seam
  introduced. Every future map UI widget test reuses it rather than rebuilding scaffolding.
- **The floor controller stand-in is a hand-written fake**, not a generated mock. The tests
  need to drive state changes — change the available floors, change the current floor,
  notify — and assert the widget reacts; doing that through a generated mock of a change
  notifier is awkward, and several of the required tests do exactly that.
- **Modules tested:** the horizontal floor switcher in both sizes. The vertical variant is
  not newly covered; its rename is mechanical.
- **Behaviours covered:** expanding and collapsing via the collapsed circle and via the
  close icon; the icon showing when no floor is selected and the floor label showing when
  one is; selecting a floor both switching to it and collapsing; re-selecting the current
  floor deselecting and returning to the icon state; rendering nothing when no floors are
  available; a value-differing floor set resetting expansion while an equal but rebuilt one
  does not; and the floor label builder being applied, on both variants.
- **Prior art:** none in this repository. There are no widget tests today; the existing
  suite is entirely pure-Dart unit tests over controllers and models, and the test suite's
  migration to the Flutter test framework is a separate, prerequisite piece of work. The
  closest usable pattern is the SBB design system package's harness, which mounts
  components inside a themed application wrapper.
- **Coverage exclusions are lifted for the new files only.** The vertical switcher's files
  keep their existing exclusion; widening coverage there is a separate cleanup. Expect the
  project's coverage figure to move on the first change.
- **No golden tests, by decision.** Layout, spacing and colour regressions will not be
  caught automatically. This is a deliberate trade against standing up golden
  infrastructure the repository does not have, on CI that runs the suite across the whole
  Flutter SDK support window on Linux, where pinned reference images are fragile.

## Out of Scope

- **Golden tests.** See above; the consequence is accepted, not overlooked.
- **Accessibility annotations.** The package has none anywhere today. A collapsible control
  is meaningfully worse for screen readers than an always-expanded list, since the floor
  list is absent from the widget tree while collapsed. Worth revisiting before 3.0.0 makes
  the horizontal variant the default, but not addressed here.
- **Overflow with many floors.** With enough floors the pill grows past the left edge of
  the screen. Deliberately not handled and deliberately not documented.
- **A configurable expansion direction.** The control is fixed as right-anchored and
  left-growing. Revisit only if a consumer asks; a direction setting would double the
  layout and animation cases with no design behind it.
- **Any visual change to the vertical floor switcher.** It is renamed and gains the floor
  label builder; nothing else.
- **Consolidating the four widget classes behind a size parameter.** A candidate for 3.0.0,
  not for this work.
- **Localization inside the package.** Floor naming is the consumer's, via the label
  builder.
- **Collapse triggered by map interaction or by a timeout.** Neither is included: a timeout
  is unpredictable, and collapsing on map gestures would require new plumbing from the map
  into the UI container.

## Further Notes

- **Worth a check with design before this closes:**
  - The floors icon renders roughly one control's height below the style switcher's own
    layers icon in the default UI — two near-identical layer glyphs, stacked, meaning
    different things. The design frame shows the switcher in isolation, so this adjacency
    was never visible in review.
  - The default UI's spacing and alignment for the horizontal pill, once it is running in
    the example application. Same reason.
  - Whether the design specifies an expand duration. If it does, it overrides the value
    chosen here for consistency with the existing tile animation.
- **The design frame is partly superseded.** It shows a floor label in a circle on the
  right of the expanded pill; the agreed interaction replaces that with a close icon. The
  state table above is authoritative.
- **On "no floor selected" versus ground level:** the controller represents no selection as
  an absent value, which resets the map's layer filters to the ground level — so no
  selection and ground level produce identical map state but are distinct UI states. The
  horizontal variant is the first control able to show the difference.
